packages/flutter/lib/src/widgets/navigator.dart

```diff
@@ -4078,7 +4078,7 @@ class NavigatorState extends State<Navigator> with TickerProviderStateMixin {
   @override
   Widget build(BuildContext context) {
     assert(!_debugLocked);
-    assert(_history.isNotEmpty);
+    assert(_history.isNotEmpty, 'You may have accidentally called Navigator.pop() in the wrong context.');
```

Comment:
This requires a test, you can in the test try to pop navigator to empty and catch the assertion error and verify its string matches expectation

Reply:
I've added a test for this. Please take a look. Thank you! :)

Reply:
Looking at the error message again it feels a bit confusing. There are a lot of ways user can end up in this situation, and poping the wrong context is only one of them. If user arrive this assertion in a different way, this message can be misleading. It is too specific, I don't have a good idea on the top of my head too...

Reply:
Hmm, I agree that this message is too specific.
```suggestion
    assert(_history.isNotEmpty, 'There is no navigation history in the current context. You may need to check Navigator routes.');
```
This is the only thing came into my mind... I wonder if this message would be helpful to users 🤔